### DISCUSSION TITLE
Is "django_otp.plugins.otp_totp" a requirement for django_otp

### DISCUSSION THREAD

**Discussion by Dennis-Asamoah** [django-otp/django-otp#171] [Q&A]
hi,
Whenever i try to remove "django_otp.plugins.otp_totp", from the installed app, 
i get an error when i try to login.
(OperationalError at /account/login/
no such table: django_otp_totpdevice
)
Even though i excluded the totp  plugin and rightfully so no totp table was migrated.
My question is, is "django_otp.plugins.otp_totp" required for django_otp to work?

the idea is  to exclude totp from my 2fa.

**Comment by psagers** [comment]
It certainly shouldn't be required. A full stack trace would no doubt shed some light on the issue.

**Reply by Dennis-Asamoah** [reply]
https://github.com/django-otp/django-otp/blob/4137746b1db701b49799f07d5d7e16be4c63ef84/src/django_otp/__init__.py#L160

```
for config in apps.get_app_configs():
        for model in config.get_models():
            if issubclass(model, Device) and not model._meta.proxy:
                yield model
```
a breakpoint shows that " model in config.get_models()", contains the ToTpDevice, which is strange as it doesnt exist in my installed app. Also i just wanted to experiment with django_otp before integrating it in my oroject, and thus the the only installed apps are 

"django_otp",
"django_otp.plugins.otp_static",
"two_factor"

putting "django_otp.plugins.otp_totp", back and migrating fixes the problem.
the path account/login breaks if not included.
